generators/Reporting.fs

```diff
@@ -0,0 +1,92 @@
+module Generators.Reporting
+
+open Serilog
+open CanonicalData
+open Exercise
+open Options
+       
+let private isOutdated (exercise: GeneratorExercise) options (parseCanonicalData':string -> CanonicalData) =
+    let cData = parseCanonicalData' exercise.Name
+    match cData.Version,exercise.ReadVersion() with
+    | canonVersion,exerciseVersion when canonVersion.Equals exerciseVersion -> false
+    | _,_ -> true
+    
+let private filterByStatus options (parseCanonicalData':string -> CanonicalData) (exercise: Exercise) =
+    match options.Status, exercise with
+    | None, _ -> true
+    | Some Status.Implemented,   Exercise.Generator _     -> true
+    | Some Status.Unimplemented, Exercise.Unimplemented _ -> true
+    | Some Status.MissingData,   Exercise.MissingData _   -> true
+    | Some Status.Deprecated,    Exercise.Deprecated _    -> true
+    | Some Status.Custom,        Exercise.Custom _        -> true
+    | Some Status.Outdated,      Exercise.Generator exercise when isOutdated exercise options parseCanonicalData' -> true
+    | Some Status.All,           _                        -> true
+    | _ -> false
+    
+type SummaryTypes =
+    | Custom of string
+    | Unimplemented of string
+    | MissingData of string
+    | Deprecated of string
+    | UpToDate of string
+    | Outdated of string * string * string
+    
+let nameFromSummaryType = function
+    | Custom n -> n
+    | Unimplemented  n -> n
+    | MissingData n -> n
+    | Deprecated n -> n
+    | UpToDate n -> n
+    | Outdated (n,_,_) -> n
+    
+let private summarizeExercise options (parseCanonicalData':string -> CanonicalData) (exercise:Exercise)  =
+
+    match exercise with
+    | Exercise.Custom custom ->                 SummaryTypes.Custom custom.Name
+    | Exercise.Unimplemented unimplemented ->   SummaryTypes.Unimplemented unimplemented.Name
+    | Exercise.MissingData missingData ->       SummaryTypes.MissingData missingData.Name
+    | Exercise.Deprecated deprecated ->         SummaryTypes.Deprecated deprecated.Name
+    | Exercise.Generator generator ->
+        let cData = parseCanonicalData' generator.Name
+        
+        match generator.ReadVersion (), cData.Version with
+        | a,b when a.Equals b ->  UpToDate generator.Name
+        | a,b -> Outdated (generator.Name,a,b)
+        
+let listExercises options =
+    Log.Information(sprintf "Listing exercises with status %s..." (options.Status.Value.ToString()))
+    
+    let parseCanonicalData' = CanonicalData.parseCanonicalData options
+    
+    let allExercises = createExercises options
+    
+    let longestNameLength = allExercises |> List.map exerciseName |> List.map (fun e -> e.Length) |> List.max
+    let indentAfter (name:string) = String.replicate (longestNameLength+2-name.Length) " "
+    
+    let exerciseGroups = allExercises
```

Comment:
Could you slightly reformat this to have the pipes start at the same level as the expression that is being piped into it?

```fsharp
let exerciseGroups = 
    allExercises
    |> List.filter (filterByStatus options parseCanonicalData')
    |> List.map (summarizeExercise options parseCanonicalData')
    |> List.groupBy (fun x -> x.GetType())
```